test/integration_test/execute_rule/insufficient_tokenholder_funds.js

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @dev This is the integration test for failing the executeRule when tokenholder
+ *      has insufficient balance.
+ *      In the tests,TH has 200 BTs. But,transfer of 250 tokens
+ *      is initiated. It will fail as TH doesn't have 250 tokens.
+ *
+ *        Following steps are performed in the test :-
+ *
+ *        - EIP20TokenMock contract is deployed.
+ *        - Organization contract is deployed and worker is set.
+ *        - TokenRules contract is deployed.
+ *         TransferRule contract is deployed and it is registered in TokenRules.
+ *        - TokenHolder contract is deployed by providing the wallets and
+ *           required confirmations.
+ *        - Using EIP20TokenMock's setBalance method,tokens are provided to TH.
+ *        - We generate executable data for TransferRule contract's transferFrom
+ *           method.
+ *        - Relayer calls executeRule method of tokenholder contract.
+ *           After it's execution below verifications are done:
+ *            - RuleExecuted event.
+ *            - tokenholder balance.
+ *            - 'to' address balance.
+ */
+const EthUtils = require('ethereumjs-util'),
+  utils = require('../../test_lib/utils'),
+  AccountsProvider = utils.AccountProvider,
+  ExecuteRuleUtils = require('./utils'),
+  BN = require('bn.js'),
+  { Event } = require('../../test_lib/event_decoder');
+
+contract('TokenHolder::executeRule', async (accounts) => {
+
+  let accountProvider,
+    tokenHolder,
+    wallet1,
+    eip20TokenMock,
+    transferRule,
+    tokenRules,
+    ephemeralPrivateKey1,
+    ephemeralKeyAddress1,
+    keyData,
+    totalBalance = 200,
+    worker;
+
+  describe('ExecuteRule integration test', async () => {
+
+    it('Should fail when tokenholder has insufficient balance', async () => {
```

Comment:
Below it statement IMO is more clear:

"Verifies that execution status is false when tokenholder has insufficient balance"